studio/components/ui/SchemaSelector.tsx

```diff
@@ -0,0 +1,169 @@
+import { useState } from 'react'
+import {
+  AlertDescription_Shadcn_,
+  AlertTitle_Shadcn_,
+  Alert_Shadcn_,
+  Button,
+  CommandEmpty_Shadcn_,
+  CommandGroup_Shadcn_,
+  CommandInput_Shadcn_,
+  CommandItem_Shadcn_,
+  CommandList_Shadcn_,
+  Command_Shadcn_,
+  IconCheck,
+  IconCode,
+  IconLoader,
+  IconPlus,
+  PopoverContent_Shadcn_,
+  PopoverTrigger_Shadcn_,
+  Popover_Shadcn_,
+  ScrollArea,
+} from 'ui'
+
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import { useSchemasQuery } from 'data/database/schemas-query'
+
+interface SchemaSelectorProps {
+  size?: 'tiny' | 'small'
+  showError?: boolean
+  selectedSchemaName: string
+  onSelectSchema: (name: string) => void
+  onSelectCreateSchema?: () => void
+}
+
+const SchemaSelector = ({
+  size = 'tiny',
+  showError = true,
+  selectedSchemaName,
+  onSelectSchema,
+  onSelectCreateSchema,
+}: SchemaSelectorProps) => {
+  const [open, setOpen] = useState(false)
+
+  const { project } = useProjectContext()
+  const {
+    data,
+    isLoading: isSchemasLoading,
+    isSuccess: isSchemasSuccess,
+    isError: isSchemasError,
+    error: schemasError,
+    refetch: refetchSchemas,
+  } = useSchemasQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+  })
+
+  const schemas = (data ?? []).sort((a, b) => (a.name > b.name ? 0 : -1))
+
+  return (
+    <div>
```

Comment:
there's also another `<div>` surrounding this `<div` in studio/components/layouts/TableEditorLayout/TableEditorMenu.tsx

Reply:
added className prop to SchemaSelector